src/include/api.h

```diff
@@ -86,6 +86,7 @@
     do {                                                                     \
         bool __autotxn = false, __update = false;                            \
         API_CALL(s, h, n, bt, config, cfg);                                  \
+        WT_ERR(__wt_txn_err_chk(s));                                         \
```

Comment:
To be clear in my understanding, this piece, among all these other changes, really is the core of the change. This function and the two calls here.

Reply:
This is the core of the change originally considered in WT-5324, to try and catch the failure sooner. In looking at our WT_TXN_ERROR handling, I came across some problems, and they were intertwined enough that I didn't think creating new tickets would help.